src/System.IO.Compression/tests/DeflateStreamAsyncTests.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    internal class ManualSyncMemoryStream : MemoryStream
+    {
+        public static async Task<ManualSyncMemoryStream> GetStreamFromFileAsync(String testFile, bool sync = false, bool strip = false)
+        {
+            var baseStream = await StreamHelpers.CreateTempCopyStream(testFile);
+            if (strip)
+            {
+                baseStream = StripHeaderAndFooter.Strip(baseStream);
+            }
+
+            var ms = new ManualSyncMemoryStream(sync);
+            await baseStream.CopyToAsync(ms);
+
+            ms.Position = 0;
+            return ms;
+        }
+
+        public ManualResetEventSlim manualResetEvent = new ManualResetEventSlim(false);
+
+        private bool isSync = false;
+        public ManualSyncMemoryStream(bool sync) : base()
+        {
+            isSync = sync;
+        }
+
+        public void SetManualEvent()
+        {
+            manualResetEvent.Set();
+            // Logger.LogInformation("Signal");
+        }
+
+        public void ResetManualEvent()
+        {
+            manualResetEvent.Reset();
+        }
+
+        public override async Task<int> ReadAsync(Byte[] array, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (isSync)
+            {
+                manualResetEvent.Wait();
+            }
+            else
+            {
+                await Task.Run(() => manualResetEvent.Wait());
+            }
+ 
+            return await base.ReadAsync(array, offset, count, cancellationToken);
+        }
+
+        public override async Task WriteAsync(Byte[] array, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (isSync)
+            {
+                manualResetEvent.Wait(cancellationToken);
+            }
+            else
+            {
+                await Task.Run(() => manualResetEvent.Wait(cancellationToken));
+            }
+
+            await base.WriteAsync(array, offset, count, cancellationToken);
+        }
+    }
+
+    public partial class DeflateStreamTests
```

Comment:
It seems like the tests in this file were separated out because they're async-related, but there are a bunch of async-related tests in the DeflateStreamTests.cs file... why are these separate?